ext/MatrixAlgebraKitGenericExt.jl

```diff
@@ -0,0 +1,230 @@
+        return USVᴴ
+    end
+    S̃ = zeros(eltype(S), size(S))
+    U_compact, S_compact, V_compact = GenericLinearAlgebra.svd(A)
+    S̃[1:minmn, 1:minmn] .= Diagonal(S_compact)
+    Ũ = _gram_schmidt(U_compact)
+    Ṽ = _gram_schmidt(V_compact)
+
+    copyto!(U, Ũ)
+    copyto!(S, S̃)
+    copyto!(Vᴴ, Ṽ')
+
+    return MatrixAlgebraKit.gaugefix!(svd_full!, U, S, Vᴴ, m, n)
+end
+
+function MatrixAlgebraKit.svd_vals!(A::AbstractMatrix{T}, S, alg::BigFloat_svd_QRIteration) where {T <: Union{BigFloat, Complex{BigFloat}}}
+    check_input(svd_vals!, A, S, alg)
+    S̃ = GenericLinearAlgebra.svdvals!(A)
+    copyto!(S, S̃)
+    return S
+end
+
+function MatrixAlgebraKit.default_eig_algorithm(::Type{T}; kwargs...) where {T <: StridedMatrix{<:Union{BigFloat, Complex{BigFloat}}}}
+    return BigFloat_eig_Francis(; kwargs...)
+end
+
+function MatrixAlgebraKit.eig_full!(A::AbstractMatrix{T}, DV, alg::BigFloat_eig_Francis)::Tuple{Diagonal{Complex{BigFloat}}, Matrix{Complex{BigFloat}}} where {T <: Union{BigFloat, Complex{BigFloat}}}
+    D, V = DV
+    D̃, Ṽ = GenericSchur.eigen!(A)
+    copyto!(D, Diagonal(D̃))
+    copyto!(V, Ṽ)
+    return D, V
+end
+
+function MatrixAlgebraKit.eig_vals!(A::AbstractMatrix{T}, D, alg::BigFloat_eig_Francis)::Vector{Complex{BigFloat}} where {T <: Union{BigFloat, Complex{BigFloat}}}
+    check_input(eig_vals!, A, D, alg)
+    eigval = GenericSchur.eigvals!(A)
+    return eigval
+end
+
+
+function MatrixAlgebraKit.default_eigh_algorithm(::Type{T}; kwargs...) where {T <: StridedMatrix{<:Union{BigFloat, Complex{BigFloat}}}}
+    return BigFloat_eigh_Francis(; kwargs...)
+end
+
+function MatrixAlgebraKit.eigh_full!(A::AbstractMatrix{T}, DV, alg::BigFloat_eigh_Francis)::Tuple{Diagonal{BigFloat}, Matrix{T}} where {T <: Union{BigFloat, Complex{BigFloat}}}
+    check_input(eigh_full!, A, DV, alg)
+    eigval, eigvec = GenericLinearAlgebra.eigen(A; sortby = λ -> real(λ))
+    return Diagonal(eigval), eigvec
+end
+
+function MatrixAlgebraKit.eigh_vals!(A::AbstractMatrix{T}, D, alg::BigFloat_eigh_Francis)::Vector{BigFloat} where {T <: Union{BigFloat, Complex{BigFloat}}}
+    check_input(eigh_vals!, A, D, alg)
+    D = GenericLinearAlgebra.eigvals(A; sortby = λ -> real(λ))
+    return real.(D)
+end
+
+function MatrixAlgebraKit.default_qr_algorithm(::Type{T}; kwargs...) where {T <: StridedMatrix{<:Union{BigFloat, Complex{BigFloat}}}}
+    return BigFloat_QR_Householder(; kwargs...)
+end
+
+function MatrixAlgebraKit.qr_full!(A::AbstractMatrix, QR, alg::BigFloat_QR_Householder)
+    Q, R = QR
+    m, n = size(A)
+    minmn = min(m, n)
+    computeR = length(R) > 0
+
+    Q_zero = zeros(eltype(Q), (m, minmn))
+    R_zero = zeros(eltype(R), (minmn, n))
+    Q_compact, R_compact = _bigfloat_householder_qr!(A, Q_zero, R_zero; alg.kwargs...)
+    copyto!(Q, _gram_schmidt(Q_compact[:, 1:min(m, n)]))
+    if computeR
+        R̃ = zeros(eltype(R), m, n)
+        R̃[1:minmn, 1:n] .= R_compact
+        copyto!(R, R̃)
```

Comment:
Can this be done directly in the correct destination?
```suggestion
        R = fill!(R, zero(eltype(R)))
        R[1:minmn, 1:n] .= R_compact
```